src/DnsHelperUI/MainForm.cs

```diff
@@ -225,6 +239,15 @@ private List<DnsProvider> LoadJsonData(string jsonFilePath)
             }
         }
 
-        
+        private void cmbNics_SelectedIndexChanged(object sender, EventArgs e)
+        {
+
+            CurrentNic = cmbNics.Items[cmbNics.SelectedIndex].ToString();
+
+            // Update the DNS values to the currently applied ones
+            var dnsServers = RefreshDNSValues();
+            SetDnsTextBoxValues(dnsServers != null && dnsServers.Length > 0 ? dnsServers[0] : string.Empty,
+                                dnsServers != null && dnsServers.Length > 1 ? dnsServers[1] : string.Empty);
```

Comment:
See my comment regarding `CurrentNic` property above. 
Remove the update functions if you implement the code that way, only assign the property value here.

I.e. this code goes away:
```
var dnsServers = RefreshDNSValues();
            SetDnsTextBoxValues(dnsServers != null && dnsServers.Length > 0 ? dnsServers[0] : string.Empty,
                                dnsServers != null && dnsServers.Length > 1 ? dnsServers[1] : string.Empty);
```